source/Ship.cpp

```diff
@@ -209,16 +218,33 @@ void Ship::Load(const DataNode &node)
 			neverDisabled = true;
 		else if(key == "uncapturable")
 			isCapturable = false;
-		else if((key == "fighter" || key == "drone") && child.Size() >= 3)
+		else if(((key == "fighter" || key == "drone") && child.Size() >= 3) ||
+			(key == "bay" && child.Size() >= 4))
 		{
+			// The drone and fighter keywords are for backwards compatiblity
+			// Initialize as if the keyword was fighter. The values will be
+			// overwritten in case of drone or bay keywords.
+			// Newer format is bay "BayCategory".
+			string category = "Fighter";
+			int childOffset = 0;
+			if(key == "drone")
+				category = "Drone";			
+			else if(key == "bay")
+			{
+				category = child.Token(1);
```

Comment:
Validation needed here - we don't support arbitrary carried types yet, so the bay category must be in BAY_TYPES.

When we do support custom bay types, the validation will have to be moved to Ship::FinishLoading as there won't be a guarantee that all bay types have been loaded yet.

Reply:
I added validation if the category is known/valid and skip loading for invalid categories.